src/sdk/pynni/nni/compression/torch/compressor.py

```diff
@@ -272,9 +268,27 @@ class Pruner(Compressor):
 
     """
 
-    def __init__(self, model, config_list):
-        super().__init__(model, config_list)
-
+    def __init__(self, model, config_list, optimizer=None):
```

Comment:
i would suggest to add one more argument to indicate whether calculate mask within __init__.

Reply:
fixed, calling compress will calculate default mask.